src/CommandLineParser.ts

```diff
@@ -0,0 +1,169 @@
+import * as ts from "typescript";
+import * as yargs from 'yargs';
+import * as fs from "fs";
+import * as path from "path";
+
+// ES6 syntax broken
+import dedent = require("dedent");
+
+export interface CompilerOptions extends ts.CompilerOptions {
+    addHeader?: boolean;
+    luaTarget?: string;
+}
+
+export interface ParsedCommandLine extends ts.ParsedCommandLine {
+    options: CompilerOptions;
+}
+
+const optionDeclarations: { [key: string]: yargs.Options } = {
+    'lt': {
+        alias: 'luaTarget',
+        choices: ['JIT', '5.3'],
+        default: 'JIT',
+        describe: 'Specify Lua target version.',
+        type: 'string'
+    },
+    'ah': {
+        alias: 'addHeader',
+        describe: 'Specify if a header will be added to compiled files.',
+        default: false,
```

Comment:
I think we should add the header by default

Reply:
Idk I think it looks cheap. Like those weird video converters youtube downloaders I used as a kid and that left their watermark everywhere.

Or is there an actual benefit in having a header that im not seeing?

Reply:
No actual benefit, just some credit/advertising for us in case people see the code in other people's repositories. Also makes it clear the code is generated.

Reply:
Yeah that acutally makes sense.
But maybe reduce the it to a one liner?
I will jsut reeanble it and we can change that later if we want to....

Reply:
I am fine with changing it to be only 1 or 2 lines. I got a nice comment from @zapp-brannigan-dota about maybe adding the version number. I would say the header should at least contain the following things: 'Generated by TypescriptToLua https://github.com/Perryvw/TypescriptToLua', the date, and the version of the transpiler that was used (should be possible right?).

Reply:
Why is the date important? With it I wouldn't add the header to any files that might end up version controlled.

Reply:
Figured date would be useful for keeping track of when files changed but I guess it will also be in the file metadata/commits so maybe we can indeed leave it out.